### PR TITLE
Upstreams and ASGs links

### PR DESCRIPTION
**Pull request by DavidIHunt**
This feature allows users to:

 - view an ASG from the Ops > Upstreams screen by clicking on of the hosts
 - see upstream status information on a new tab on the ASG details dialog

### CODE REVIEW

File: server/api/controllers/services/servicesController.js [hunk 1]
```diff
@@ -18,6 +23,36 @@ function getServices(req, res, next) {
 }
 
 /**
+ * GET /services/{service}/asgs
+ */
+function getASGsByService(req, res, next) {
```
Comment (Merlin-Taylor) [inline server/api/controllers/services/servicesController.js]: Response to non-existent environment has status code 400 and error message. Response to non-existent service or slice is an empty array.
If no slice parameter is appended to the query string, blue and green slices are filtered out of the response, which is not intuitive. Suggest either mandatory slice parameter or unfiltered response if slice parameter is not specified.

Comment (DavidIHunt) [inline server/api/controllers/services/servicesController.js]: Yes I agree good point! I'll change that.

File: server/api/swagger.yaml [hunk 2]
```diff
@@ -812,6 +812,33 @@ paths:
...
+      parameters:
+        - name: service
+          in: path
+          type: string
+          required: true
+        - name: environment
+          in: query
+          type: string
+          required: true
+        - name: slice
```
Comment (Merlin-Taylor) [inline server/api/swagger.yaml]: An enumeration of valid stings for a slice (blue | green | none) is defined in a couple of places in the swagger doc. Suggest extracting this as its own definition.

**Comment by duncanhall** [comment]
@DavidIHunt - is this still 'unmergeable'? Should we close PRs if they are not ready to be merged?

**Pull request merged**